main.go

```diff
@@ -231,7 +231,12 @@ func wrappedMain() int {
 
 func cliConfigFile() (string, error) {
 	mustExist := true
-	configFilePath := os.Getenv("TERRAFORM_CONFIG")
+
+	configFilePath := os.Getenv("TF_CLI_CONFIG_FILE")
+	if configFilePath == "" {
+		configFilePath = os.Getenv("TERRAFORM_CONFIG")
```

Comment:
I'm not sure (from the discussion here) if we're after all deprecating this variable, but if we are - may I suggest to at least log `WARN` and add an appropriate comment here about the deprecation, and perhaps suggest to later (in future release) go as far as erroring out if this variable is not empty?

Reply:
I think TFE (or rather the terraform build worker) relies on this, so adding a `WARN` would probably spam TFE users with a warning they can't do anything about. I'm mostly in favor of deprecating the old name, but maybe that should be a separate PR that loops in the relevant TFE people! (Looking at blame for the spot that uses that, that's mostly @ryanuber and possibly extra advice from @phinze and @evanphx.)

Reply:
I agree that deprecation should be a separate pull request. Most users won't know that the `TERRAFORM_CONFIG` environment variable even exists without looking at the code, this pull request, or the linked issue. I think the documentation updates will show users that they should now be using `TF_CLI_CONFIG_FILE` instead. Once we do deprecate the variable, I do agree that we should add a `WARN` or something similar but I'm not sure what the implications are going to be so I'll defer to others.